show/plugins/barefoot.py

```diff
@@ -22,24 +22,32 @@ def profile():
     hwsku_dir = device_info.get_path_to_hwsku_dir()
     with open(hwsku_dir + '/switch-tna-sai.conf') as file:
         chip_family = json.load(file)['chip_list'][0]['chip_family'].lower()
-    
+
     # Print current profile
     click.echo('Current profile: ', nl=False)
     subprocess.run('docker exec -it syncd readlink /opt/bfn/install | sed '
-        r's/install_\\\(.\*\\\)_profile/\\1/', check=True, shell=True)
-    
-    # Exclude current and unsupported profiles
+        r's/install_\\\(.\*\\\)_profile/\\1/'
+        r' | sed s/install_\\\(.\*\\\)_tofino\\\(.\*\\\)/\\1/', check=True, shell=True)
+
+    # Check if profile naming format contains tofino family information 
+    output = subprocess.check_output(['docker', 'exec', '-it', 'syncd', 'ls', '/opt/bfn']).strip().decode()
+    suffix = '_' + chip_family if '_tofino' in output else '_profile'
+
+    # Check supported profiles 
     opts = ''
     if chip_family == 'tofino':
-        opts = r'\! -name install_y\*_profile '
+        opts = r' -name install_x\*' + suffix
     elif chip_family == 'tofino2':
-        opts = r'\! -name install_x\*_profile '
-    
+        opts = r' -name install_y\*' + suffix
+    else:
+        opts = r' -name \*' + suffix
+
     # Print profile list
     click.echo('Available profile(s):')
     subprocess.run('docker exec -it syncd find /opt/bfn -mindepth 1 '
-        r'-maxdepth 1 -type d -name install_\*_profile ' + opts + '| sed '
-        r's%/opt/bfn/install_\\\(.\*\\\)_profile%\\1%', shell=True)
+        r'-maxdepth 1 -type d,l ' + opts + '| sed '
+        r's%/opt/bfn/install_\\\(.\*\\\)_profile%\\1%'
+        r' | sed s%/opt/bfn/install_\\\(.\*\\\)_tofino\\\(.\*\\\)%\\1%', shell=True)
```

Comment:
The second capture group can be omitted.

Reply:
@oleksandrx-kolomeiets, it really looks better, thank you!